test/io.test.ts

```diff
@@ -149,21 +149,11 @@ it('modifies incoming server event', async () => {
       connection.server.send(event.data)
     })
 
-    io.server.on('greeting', (message) => {
+    io.server.on('greeting', function (message) {
```

Comment:
Another downside: diverging experience between using the raw WebSocket interceptor and the wrapper. 

## Proposal

Maybe always set the `event` instance first, and spread the rest of the data in the remaining args?

```js
server.on('message', (event, ...data) => {})
client.on('hello', (event, ...data) => {}
```

**Pros:**

- No function binding; 
- Identical experience between the raw interceptor and the wrapper;

**Cons:**

- Diverging experience between the wrapper and the Socket.IO client.